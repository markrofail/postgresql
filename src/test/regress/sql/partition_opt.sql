create table bt (k integer, v integer) partition by range (k);
create table dt1 partition of bt for values from (1) to (10001);
create table dt2 partition of bt for values from (10001) to (20001);
create index dti1 on dt1(v);
create index dti2 on dt2(v);
insert into bt values (generate_series(1,20000), generate_series(1,20000));
analyze bt;
explain (costs off) select * from bt where k between 1 and 20000 and v = 100;

CREATE TABLE measurement (
    city_id         int not null,
    logdate         date not null,
    peaktemp        int,
    unitsales       int
) PARTITION BY RANGE (logdate);


CREATE TABLE measurement_y2006m03 PARTITION OF measurement
    FOR VALUES FROM ('2006-03-01') TO ('2006-04-01');

explain (costs off) select * from measurement where logdate between '2006-03-01' AND '2006-03-31';

drop table bt;
drop table measurement;
