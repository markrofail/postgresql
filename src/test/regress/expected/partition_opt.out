create table bt (k integer, v integer) partition by range (k);
create table dt1 partition of bt for values from (1) to (10001);
create table dt2 partition of bt for values from (10001) to (20001);
create index dti1 on dt1(v);
create index dti2 on dt2(v);
insert into bt values (generate_series(1,20000), generate_series(1,20000));
analyze bt;
explain select * from bt where k between 1 and 20000 and v = 100;
                              QUERY PLAN                              
----------------------------------------------------------------------
 Append  (cost=0.29..16.61 rows=2 width=8)
   ->  Index Scan using dti1 on dt1  (cost=0.29..8.30 rows=1 width=8)
         Index Cond: (v = 100)
   ->  Index Scan using dti2 on dt2  (cost=0.29..8.30 rows=1 width=8)
         Index Cond: (v = 100)
(5 rows)

CREATE TABLE measurement (
    city_id         int not null,
    logdate         date not null,
    peaktemp        int,
    unitsales       int
) PARTITION BY RANGE (logdate);
CREATE TABLE measurement_y2006m03 PARTITION OF measurement
    FOR VALUES FROM ('2006-03-01') TO ('2006-04-01');
explain select * from measurement where logdate between '2006-03-01' AND '2006-03-31';
                                  QUERY PLAN                                   
-------------------------------------------------------------------------------
 Append  (cost=0.00..28.50 rows=1850 width=16)
   ->  Seq Scan on measurement_y2006m03  (cost=0.00..28.50 rows=1850 width=16)
(2 rows)

drop table bt;
drop table measurement;
